keps/0034-instance-health.md

```diff
@@ -0,0 +1,43 @@
+---
+kep-number: 34
+short-desc: Provide a way to monitor health of KUDO instance post-deployment
+title: KUDO instance health
+authors:
+  - "@alenkacz"
+owners:
+  - "@alenkacz"
+editor: TBD
+creation-date: 2020-09-24
+last-updated: 2020-09-24
+status: provisional
+---
+
+# KUDO instance health
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+KUDO helps people implement their operators and it's focus is day 2 operations. Part of day 2 is also monitoring your workload health after deployment. To help with that, KUDO will expose "health" computed as a heuristic based on health and readiness of the underlying resources. In the first iteration, health will be just a simple heuristic computed from Pods, StatefulSets, Deployments, ReplicaSets, DaemonSets and Services (let's call them *health phase 1 resources*).
+
+## Motivation
+
+Added health monitoring for `Instance` CRD will help people answer question "is my operator working at this point in time" without querying all underlying resources. KUDO will expose this heuristic as part of `Status` field for everyone to query. This could be used as a signal for a montioring tool.
```

Comment:
```suggestion
Added health monitoring for `Instance` CRD will help people answer the question "is my operator working at this point in time" without querying all underlying resources. KUDO will expose this heuristic as part of `Status` field for everyone to query. This could be used as a signal for a monitoring tool.
```